todoapp/app/src/androidTestMock/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsScreenTest.java

```diff
@@ -59,30 +60,31 @@
      */
     @Rule
     public ActivityTestRule<StatisticsActivity> mStatisticsActivityTestRule =
-            new ActivityTestRule<>(StatisticsActivity.class, true, false);
+            new ActivityTestRule<>(StatisticsActivity.class, false, false);
 
-    /**
-     * Setup your test fixture with a fake task id. The {@link TaskDetailActivity} is started with
-     * a particular task id, which is then loaded from the service API.
-     *
-     * <p>
-     * Note that this test runs hermetically and is fully isolated using a fake implementation of
-     * the service API. This is a great way to make your tests more reliable and faster at the same
-     * time, since they are isolated from any outside dependencies.
-     */
     @Before
-    public void intentWithStubbedTaskId() {
-        // Given some tasks
-        ViewModelFactory.destroyInstance();
-        TasksRepository tasksRepository = Injection.provideTasksRepository(
-                InstrumentationRegistry.getTargetContext());
-        tasksRepository.deleteAllTasks();
-        tasksRepository.saveTask(new Task("Title1", "", false));
-        tasksRepository.saveTask(new Task("Title2", "", true));
-
-        // Lazily start the Activity from the ActivityTestRule
-        Intent startIntent = new Intent();
-        mStatisticsActivityTestRule.launchActivity(startIntent);
+    public void prepareRepository() throws Throwable {
+        mStatisticsActivityTestRule.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                TasksRepository repo = ViewModelFactory.getInstance(
+                        (Application) ApplicationProvider.getApplicationContext())
+                        .getTasksRepository();
+
+                repo.deleteAllTasks();
+                repo.saveTask(
```

Comment:
Could be a single line per `saveTask` call.

Reply:
Done